### DISCUSSION TITLE
Performance metrics and connection status

### DISCUSSION THREAD

**Discussion by burtaverde** [qdm12/gluetun#820] [Ideas]
Hello. I'm not sure if this is interesting to all, but it was quite useful to me so i added a few things around Gluetun to make it better. Other than the obvious setting up transmission and jackett to use GluetonVPN's network stack that way they're behind the vpn. I've also passed a firefox container that way i can go on ipleak.net to make sure the connection is working and speedtest-tracker to run automatic speed tests every hour that way i know what kind of performance to expect. Finally, this may be useless, but i have an extra script running every 10 minutes comparing transmission's IP with the server's making sure they aren't identical. If they are it stops it to prevent any long unprotected download periods and logs the time when it was detected.

**Comment by qdm12** [comment]
Hey there thanks for sharing.

Already built-in gluetun are:

- ping github.com every 5s to check the connection is still up
- public IP getter (for logs only). In the past I was using it to compare with the VPN server IP address but it turns out that it is sometimes different and that it changes over time. So this has been removed. Anyway, with the firewall this is not really possible for it to leak so there is not much point comparing IPs.

I'm planning on adding some network metrics with Prometheus sometime, although this is probably already feasible today by scrapping the Docker daemon metrics I think.

Also automatic speedtests would be interesting to be baked in future prometheus metrics

**Comment by burtaverde** [comment]
i'm not completely sure how dockers work when it comes to adding stuff, but it should be possible to make one with speedtest-tracker included right? It would take care of making nice graphs and all the testing stuff, although not include the current usage.
